### PR TITLE
bahar/Replace connect method with use store hook for no balance

### PR DESCRIPTION
**Pull request by bahar-sadeghzadeh**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   ...

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by farzin-deriv — APPROVED** [review]

File: packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx [hunk 1]
```diff
@@ -4,21 +4,32 @@ import { createBrowserHistory } from 'history';
...
-jest.mock('Stores/connect', () => ({
-    __esModule: true,
-    default: 'mockedDefaultExport',
-    connect: () => Component => Component,
-}));
+import { StoreProvider } from '../../../hooks';
 
 describe('<NoBalance />', () => {
     const history = createBrowserHistory();
+    let mockRootStore;
```
Comment (hamid-deriv) [inline packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx:11]:
```suggestion
    let mockRootStore: DeepPartial<TRootStore>;
```

File: packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx [hunk 2]
```diff
@@ -4,21 +4,32 @@ import { createBrowserHistory } from 'history';
...
+            },
+        };
+    });
 
     it('component should render', () => {
         render(
             <Router history={history}>
-                <NoBalance is_deposit_locked={false} />
-            </Router>
+                <NoBalance />
```
Comment (hamid-deriv) [inline packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx:28]: Suggestion: As this render argument is identical for all test cases, can it be a function used in test cases?

**Review by hamid-deriv — APPROVED** [review]

File: packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx (hunk 2 above)
Comment (bahar-sadeghzadeh) [reply packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx:28]: @hamid-deriv I think in this way we'll lose the readability of the code, so let it stay like this

File: packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx (hunk 1 above)
Comment (bahar-sadeghzadeh) [reply packages/cashier/src/components/no-balance/__tests__/no-balance.spec.tsx:11]: @hamid-deriv By assigning the type I won't be able to make any change inmockRootStore hence I need to pass new amounts during the test, so I didn't assign type here

**Pull request merged**